File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.js
var zigzagLevelOrder = function(root) {
     if(!root) return [];
    let queue = [root];
    let levels = [];
    while(queue.length > 0) {
        let levelSize = queue.length;
        levels.push([]);
        for(let i = 0; i < levelSize; i++) {
            let curr = queue.shift();
            if(levels.length % 2 == 1) levels[levels.length - 1].push(curr.val);
            else levels[levels.length - 1].unshift(curr.val);  
            
            if(curr.left) queue.push(curr.left);
            if(curr.right) queue.push(curr.right);
        }
    }
    
    return levels;
};

File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.py
class Solution(object):
    def zigzagLevelOrder(self, root):
        levels=[]
        if not root:
            return []
        queue = [root]
        level=0
        while queue:
            rev = []
            for i in range(len(queue)):
                curr = queue.pop(0)
                rev.append(curr.val)
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            if level%2:
                levels.append(rev[::-1])
            else: 
                levels.append(rev)
            level += 1
        return levels